src/Microsoft.SqlTools.ServiceLayer/AzureBlob/BlobService.cs

```diff
@@ -0,0 +1,123 @@
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        /// <summary>
+        /// Default, parameterless constructor.
+        /// </summary>
+        internal BlobService()
+        {
+        }
+
+        /// <summary>
+        /// Gets the singleton instance object
+        /// </summary>
+        public static BlobService Instance
+        {
+            get { return instance.Value; }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+
+            // Create shared access secret
+            serviceHost.SetRequestHandler(CreateSasRequest.Type, HandleCreateSasRequest);
+        }
+
+        internal async Task HandleCreateSasRequest(
+           CreateSasParams optionsParams,
+           RequestContext<CreateSasResponse> requestContext)
+        {
+            try
+            {
+                ConnectionInfo connInfo;
+                bool supported = IsBackupRestoreOperationSupported(optionsParams.OwnerUri, out connInfo);
+                var response = new CreateSasResponse();
+
+                if (supported && connInfo != null)
+                {
+                    SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "Backup");
+                    // Connection gets discounnected when backup is done
+                    ServerConnection serverConnection = new ServerConnection(sqlConn);
+                    Server sqlServer = new Server(serverConnection);
+
+                    SharedAccessSignatureCreator sharedAccessSignatureCreator = new SharedAccessSignatureCreator(sqlServer);
+                    string sharedAccessSignature = sharedAccessSignatureCreator.CreateSqlSASCredential(optionsParams.StorageAccountName, optionsParams.BlobContainerKey, optionsParams.BlobContainerUri, optionsParams.ExpirationDate);
+                    response.SharedAccessSignature = sharedAccessSignature;
+                }
+                await requestContext.SendResult(response);
+            }
+            catch (Exception ex)
+            {
+                await requestContext.SendError(ex.ToString());
+            }
+        }
+
+        private bool IsBackupRestoreOperationSupported(string ownerUri, out ConnectionInfo connectionInfo)
+        {
+            SqlConnection sqlConn = null;
+            try
+            {
+                ConnectionInfo connInfo;
+                BlobService.ConnectionServiceInstance.TryFindConnection(
+                        ownerUri,
+                        out connInfo);
+
+                if (connInfo != null)
+                {
+                    using (sqlConn = ConnectionService.OpenSqlConnection(connInfo, "AzureBlob"))
```

Comment:
I don't think you need to open the connection, you are not using it, you only checked the connInfo.IsCloud.

Reply:
OpenSqlConnection can return null